applications/GeoMechanicsApplication/tests/cpp_tests/test_reset_displacement_process.cpp

```diff
@@ -0,0 +1,158 @@
+// KRATOS___
+//     //   ) )
+//    //         ___      ___
+//   //  ____  //___) ) //   ) )
+//  //    / / //       //   / /
+// ((____/ / ((____   ((___/ /  MECHANICS
+//
+//  License:         geo_mechanics_application/license.txt
+//
+//  Main authors:    Richard Faasse
+//
+
+#include "custom_processes/reset_displacement_process.h"
+#include "geo_mechanics_fast_suite.h"
+
+#include <boost/numeric/ublas/assignment.hpp>
+
+namespace
+{
+
+using namespace Kratos;
+
+class StubConstitutiveLaw : public ConstitutiveLaw
+{
+};
+
+class StubElement : public Element
+{
+public:
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(StubElement);
+
+    StubElement(IndexType NewId, const GeometryType::Pointer& pGeometry) : Element(NewId, pGeometry)
+    {
+        mConstitutiveLaws = std::vector<ConstitutiveLaw::Pointer>(3, make_shared<StubConstitutiveLaw>());
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<ConstitutiveLaw::Pointer>& rVariable,
+                                      std::vector<ConstitutiveLaw::Pointer>&    rOutput,
+                                      const ProcessInfo& rCurrentProcessInfo) override
+    {
+        rOutput = mConstitutiveLaws;
+    }
+
+    void CalculateOnIntegrationPoints(const Variable<Vector>& rVariable,
+                                      std::vector<Vector>&    rOutput,
+                                      const ProcessInfo&      rCurrentProcessInfo) override
+    {
+        rOutput = mIntegrationPointVectors;
+    }
+
+    using Element::CalculateOnIntegrationPoints;
+
+    std::vector<Vector>                   mIntegrationPointVectors = {};
+    std::vector<ConstitutiveLaw::Pointer> mConstitutiveLaws        = {};
+};
+
+ModelPart& CreateModelPartWithAStubElement(Model& rModel)
+{
+    auto& model_part = rModel.CreateModelPart("MainModelPart");
+    auto  node_1     = model_part.CreateNewNode(1, 0.0, 0.0, 0.0);
+    auto  node_2     = model_part.CreateNewNode(2, 1.0, 0.0, 0.0);
+    auto  node_3     = model_part.CreateNewNode(3, 1.0, 1.0, 0.0);
+
+    auto geometry = std::make_shared<Triangle2D3<Node>>(node_1, node_2, node_3);
+    auto element  = Kratos::make_intrusive<StubElement>(1, geometry);
+    model_part.AddElement(element);
```

Comment:
We can get rid of variable `element` without loosing clarity, I think:
```suggestion
    model_part.AddElement(make_intrusive<StubElement>(1, geometry));
```

(Note that the namespace qualifier `Kratos::` is redundant here due to the `using namespace Kratos;` statement at line 21.)

Reply:
Agreed done